modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/ProgressPrinter.java

```diff
@@ -147,4 +140,15 @@ private void printProgress0(long curr, double currRatio) {
 
         printStream.print(progressBar);
     }
+
+    /**
+     * Formats the given duration in milliseconds in "HH:mm:ss" format.
+     */
+    private static String formatDuration(long durationMillis) {
```

Comment:
I think you could have used IgniteUtils.SHORT_DATE_FMT, right? What was the reason

Reply:
`DateTimeFormatter` is not suitable for formatting durations, because it creates a string representation of a point in time. Same applies for the `SimpleDateFormat`, the original code did a strange things like:
```
long timeRunning = System.currentTimeMillis() - timeStarted;
...
timeFormat.format(new Date(timeRunning));
```
Or am I missing something?

Reply:
Yes, it did exactly what you mentioned, but duration is indistinguishable from any other time instant. As far as I know, timestamps start 00:00:00 so old code was valid in that regard, it didn't use any dates.

Reply:
yes, but strictly speaking, that's a hack. However, I may be too much of a purist here, and we can convert it to an `Instant`. What do you think?

Reply:
I think you can leave it as it is right now, not a big deal. From the purist standpoint, your solution is better.